libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts

```diff
@@ -0,0 +1,312 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+    Activity,
+    Attachment,
+    CardFactory,
+    MessageFactory,
+    O365ConnectorCardActionQuery,
+    TeamsActivityHandler,
+    TurnContext
+} from 'botbuilder';
+
+//
+// You can install this bot in any scope. From the UI just @mention the bot.
+//
+export class Office365CardsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+        this.onMessage(async (context, next) => {
+            await this.sendO365CardAttachment(context);
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+            for (const member of membersAdded) {
+                if (member.id !== context.activity.recipient.id) {
+                    await context.sendActivity('Hello and welcome!');
+                }
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    }
+
+    protected async onTeamsO365ConnectorCardAction(context: TurnContext, query: O365ConnectorCardActionQuery): Promise<void> {
+        await context.sendActivity(MessageFactory.text(`O365ConnectorCardActionQuery event value: ${JSON.stringify(query)}`));
+    }
+
+    private async sendO365CardAttachment(context: TurnContext): Promise<void> {
+        const card = CardFactory.o365ConnectorCard({
+            "title": "card title",
```

Comment:
Optional: I was adding casts so at least you can refer back to the underlying interfaces. It gives some level of "documentation".

Reply:
@tsuwandy, I checked the botbuilder-core exports and it exports the O365Connector interfaces:

![o364 stuff](https://user-images.githubusercontent.com/14935595/66244174-16118d80-e6bc-11e9-872f-bf56e3c3fc35.png)

Reply:
Hmm, the schema seems to be a mismatch with the json we are generating from dotnet sample is not compatible.  Do we need to regenerate them?